ADRs/ADR0006-replace-badger-kv-store [WITHDRAWN].md

```diff
@@ -0,0 +1,225 @@
+```
+
+It also supports prefix queries.
+
+```sql
+SELECT key, value
+FROM my_table, each(hstore_column) AS h(key, value)
+WHERE key LIKE 'prefix%';
+```
+
+* Good, because it's meant as a KV store.
+* Good, because it supports prefix queries.
+* Good, because it can be indexed like any column.
+* Good, because it guarantees atomic transactions.
+* Bad, because it's Postgres specific which could impose limitations on us in the future or be a source of *unknown unknowns*.
+* Bad, because it's an extension, not a native feature, which adds complexity and maintenance of its own.
+* Bad, because it uses a novel syntax.
+* Bad, because it might not have the same performance capabilities as Badger.
+
+### [Option 1.2] - Store Keys In JSONB
+
+Option 2 would create a JSONB column with a key of TEXT and a value of JSONB. This allows more complex structures in the values column.
+
+```sql
+SELECT key, value
+FROM my_table, jsonb_each(jsonb_column) AS j(key text, value jsonb)
+WHERE key LIKE 'prefix%';
+```
+
+* Good, because it can be indexed like a normal table.
+* Good, because it can handle prefix queries.
+* Good, because it guarantees atomic transactions.
+* Good, because it easily handles more complex values.
+* Bad, because we want strict hash comparisons to continue working and JSONB interpretation could have unforeseen consequences.
+* Bad, because it might not have the same performance capabilities as Badger.
+* Bad, because it's Postgres specific which could impose limitations on us in the future or be a source of *unknown unknowns*.
+
+### [Option 1.3] - Store Keys In KV Table
+
+Option 3 would dump everything into typical TEXT types in a standard table named `transaction`.
+
+* Good, because it's simple and straight-forward table access. JSONB and HSTORE have new, Postgres specific syntax and behavior.
+* Good, because it stays within standard SQL features, and doesn't utilize Postgres specific behavior.
+* Good, because it handles prefix queries.
+* Bad, because it lacks similar performance capabilities compared to Badger.
+
+### [Option 2.1] - Make KV Store Handle Atomic Guarantees
+
+Option 2 would involve extending the `KVStore` interface so that it can be atomically controlled by the caller (i.e. TxIndexer, block store, or the state trees) either by having it directly extend the Rollback & Savepoint interface, as demonstrated below, or by having it manage its own rollbacks.
+
+```go
+func (indexer *txIndexer) Rollback([]byte) error {
+  return fmt.Errorf("not impl")
+}
+func (indexer *txIndexer) Savepoint() ([]byte, error) {
+  return nil, fmt.Errorf("not impl")
+}
+```
+
+This option is similar to [PR #657 - KV POC](https://github.com/pokt-network/pocket/pull/657) but different in a key way: it addresses the core design misalignment between the `PersistenceRWContext` and the `KVStore` causing the dissonance in the first place.
+
+* Good, because it avoids migrations.
```

Comment:
For each of the postgres options, can you add a `Bad, because it requires migrations as described above`